src/patchelf.cc

```diff
@@ -721,7 +722,7 @@ void ElfFile<ElfFileParamNames>::writeReplacedSections(Elf_Off & curOff,
                 shdr.sh_addralign = orig_shdr.sh_addralign;
 
             for (unsigned int j = 0; j < phdrs.size(); ++j)
-                if (rdi(phdrs[j].p_type) == PT_NOTE) {
+                if (rdi(phdrs[j].p_type) == PT_NOTE && noted_phdrs.find(j) == noted_phdrs.end()) {
```

Comment:
```suggestion
                if (rdi(phdrs[j].p_type) == PT_NOTE && noted_phdrs.insert(j).second) {
```

Reply:
we can't do this, until we know for sure they overlap.